core/vm/contracts.go

```diff
@@ -662,15 +664,14 @@ func (c *numberValidators) RequiredGas(input []byte) uint64 {
 }
 
 func (c *numberValidators) Run(input []byte, caller common.Address, evm *EVM, gas uint64) ([]byte, uint64, error) {
-	blockNumber := evm.Context.BlockNumber
-	validators := evm.Context.Engine.GetValidators(blockNumber, evm.Context.GetHash(blockNumber.Uint64()))
 	gas, err := debitRequiredGas(c, input, gas)
-	if err != nil {
+	if err != nil || len(input) != 0 {
```

Comment:
The length check should be separate, and return an error specific to invalid input if it fails